documentation/cubic-pos-article.md

```diff
@@ -0,0 +1,155 @@
+
+**Important parameters in the Namada PoS model**
+- Duration of a block: ~ 5 - 10 seconds
+- Duration of an epoch: ~ 1 day
+- Pipeline offset length: 2 epochs
+- Unbonding offset length: 21 epochs
+
+Below, we will review the rules around important PoS actions
+- Bonding tx submitted at epoch `n` initiates changes at epoch `n + pipeline_offset (n+2)`
+	- Bond becomes active, validator bonded stake and validator set is updated
+- Unbonding tx submitted at epoch `n`
+	- Updates validator set with decreased validator voting power at epoch `n + pipeline_offset (n+2)`
+	- Amount of the unbond is withdrawable at epoch `n + unbonding_offset + pipeline_offset (n+23)`
+		- TODO: check over this w Tomas
+- Redelegation tx submitted at epoch `n`
+	- Only support 100% token amount redelegated
+	- TODO: more details needed here perhaps?
+
+**Validator Sets**
+
+Namada validators are classified into one of three sets:
+- **Consensus:** validators that are active in the consensus voting mechanism, chosen to be the `max_validators` largest validators weighted by their voting power. They are the only validators who can earn block rewards from the PoS inflation.
+- **Below capacity:** validators whose bonded stake is less than the *consensus* validators but above some small threshold `min_validator_stake`.
+- **Below threshold:** all other validators with stake below `min_validator_stake`.
+
+The *consensus* and *below capacity* validator sets are explicitly held in storage, ordered by their bonded stake, and are updated at every epoch boundary. However, no ordered set of *below threshold* validators is kept in storage. This validator set construction and the `min_validator_stake` threshold exist primarily to prevent the possibility of significant slowing of PoS system updates due to spam validator account creation. Thus, unbounded iteration when updating and validating PoS system state changes is avoided. Conversely, the *below capacity* validator set must be kept in storage, ordered by bonded stake, in order to properly process validator set changes in which validators may drop out or enter the *consensus* validator set.
+
+**Important validator set configuration values, mutable via governance:**
+- `max_validators` = 100
+- `min_validator_stake` = 1 NAM
+
+# Cubic Slashing
+
+As in many other PoS blockchains, slashing is applied as a way to punish misbehaving validators. If a validator is detected to have misbehaved then portions of the validator's bonded stake will be slashed or seized. This propagates down to the stake of delegators associated to the misbehaving validator as well. Validator infractions that are subject to slashing include:
+
+- proposing an invalid block
+- signing an invalid block
+- proposing two different blocks at the same height
+- double-signing a block
```

Comment:
this is the same as the bullet above, no? I think we also slash for amnesia attacks, whatever Tendermint can detect.

Reply:
I thought L62 is specific to a block proposer but L63 can also apply to other validators who are just signing. Also took this list from https://specs.namada.net/economics/proof-of-stake/bonding-mechanism.html#slashing, though it's possible that this may need to be updated instead.

Not sure if Tendermint detects amnesia attacks yet, from https://github.com/tendermint/tendermint/issues/5270 it looks like it may be put into v0.37. As it stands, the only kinds of `MisbehaviorType` in tendermint are `DuplicateVote` and `LightClientAttack` (there is also an `Unknown` variant in this enum).

Reply:
hmm. we should definitely support all evidence that Tendermint can detect, and for now we can consider any sort of safety attack to be of the same sort of severity (meriting cubic slashing) (though independent parameters is fine). 

I see what you mean but I don't think there's a difference in proposing vs signing from a safety perspective, safety is < 1/3 byzantine regardless

Reply:
Aside from the amnesia attack, we currently do support all Tendermint evidences. I suppose it may depend on the Tendermint version we use as well, but https://github.com/tendermint/tendermint/issues/5270 is still open as well atm. Let me know if I should include the amnesia attack in the article.

Gotcha regarding proposing vs signing. In this case, should these 4 lines be condensed into:
```
- Signing an invalid block
- Signing two different blocks at the same height
```
?